concepts/views.py

```diff
@@ -127,30 +141,31 @@ def add_concept(request, concept_id):
         'next_page': urllib.parse.quote_plus(next_page),
         'back_to_page': back_to_page
     }
-    if concept.concept_state != Concept.PENDING:
-        return HttpResponseRedirect(next_page)
 
-    if request.GET.get('confirmed', False):
-        try:
-            manager.add()
-        except ConceptUpstreamException as E:
-            messages.error(
-                    request,
-                    'ERROR: There was an error while communicating with Conceptpower.'
-                )
-            return HttpResponseRedirect(reverse('concepts'))
-        return HttpResponseRedirect(next_page)
-
-
-    candidates = manager.get_similar()
-    matches = manager.get_matching()
-
-    context.update({
-        'candidates': candidates,
-        'matches': matches,
-    })
-
-    return render(request, "annotations/concept_add.html", context)
+    # Process only if the concept is still in a resolvable state (e.g., PENDING or FLAGGED)
+    if concept.concept_state in [Concept.PENDING, Concept.FLAGGED]:
+        if request.GET.get('confirmed', False):
```

Comment:
a GET request should not modify the state. creating a new concept should happen via a POST request.